isotovideo

```diff
@@ -89,13 +134,33 @@ sub _get_version_string () {
     return "Current version is $thisversion [interface v$OpenQA::Isotovideo::Interface::version]";
 }
 
+sub _exit_code_from_test_results() {
+    my @results = glob(path(bmwqemu::result_dir(), "/result-*.json"));
+
+    return EXIT_STATUS_ERR_NO_TESTS if scalar @results == 0;
+
+    my $did_fail = 0;
+
+    for my $result_file_path (@results) {
+        my $result_file = path(Mojo::Path->new($result_file_path)->canonicalize);
+        my $test_result = decode_json($result_file->slurp)->{result};
+        diag sprintf("Test result [%s] %s", $result_file->to_string, $test_result);
+
+        # If a failure is found, keep the failure
+        $did_fail = $test_result eq 'ok' ? 0 : 1 unless $did_fail;
```

Comment:
so this does not regard the test flags as defined in http://open.qa/docs/#_test_flags meaning that `isotovideo --exit-status-from-test-results` and openQA can come to different conclusions about the overall result if `ignore_failure` is used. Are we ok with that?

Reply:
For now I think is reasonable enough.

I can't determine if `ignore_failure` is persisted in the test result. If so then I can make it aware of it and consider that marked test module as ok instead.

Reply:
I guess it could be as simple as `decode_json($result_file->slurp)->{test_flags}` so theoretically you could do

```suggestion
        my $test = decode_json($result_file->slurp);
        $did_fail = $test->{result} eq 'ok' && $test->{test_flags} ne 'ignore_failure' ? 0 : 1 unless $did_fail;
```

(untested) pseudo code of course :)

Reply:
inspecting `basetest::save_test_result` the testflags are not persisted in the test result. Only:

```perl
my $result = {
    details => $self->details(),
    result => $self->result(),
    dents => $self->{dents},
    execution_time => $self->{execution_time},
};
$result->{extra_test_results} = $self->{extra_test_results} if $self->{extra_test_results};
```